projects/plugins/jetpack/extensions/blocks/podcast-player/edit.js

```diff
@@ -91,18 +94,28 @@ const PodcastPlayerEdit = ( {
 	const playerId = `jetpack-podcast-player-block-${ instanceId }`;
 
 	// State.
-	const [ editedUrl, setEditedUrl ] = useState( url || '' );
-	const [ isEditing, setIsEditing ] = useState( false );
-	const [ feedData, setFeedData ] = useState( exampleFeedData || {} );
 	const cancellableFetch = useRef();
-	const [ isInteractive, setIsInteractive ] = useState( false );
+	const [ { selectedGuid, checkUrl, ...state }, dispatch ] = useReducer( podcastPlayerReducer, {
+		editedUrl: url || '',
+		isEditing: ! url && ! exampleFeedData,
+		isLoading: false,
+		feedData: exampleFeedData || {},
+		isInteractive: false,
+		selectedGuid: selectedEpisodes[ 0 ]?.guid,
```

Comment:
This is causing me some issues when editing posts with old podcast players that need to be migrated. I get the following error after the migration:

```
TypeError: Cannot read property '0' of undefined
    at editor-beta.js?ver=1612870465:28
```

Reply:
That's strange as the default for `selectedEpisodes` should be an empty array. I'll run some tests!

Reply:
I've been struggling to reproduce this problem. I have noticed that we are saving the `selectedEpisodes` attribute as an empty array when that is the default value, but that shouldn't cause this problem. 

Essentially, if any part of the expression is undefined then we want `selectedGuid` to be undefined, so I added a check in 205b9515b. Could you see if the problem is still there, please?

Reply:
It seems okay now!